MovieGPT-api/src/main/java/lt/techin/group/project/service/MediaService.java

```diff
@@ -0,0 +1,96 @@
+package lt.techin.group.project.service;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lt.techin.group.project.exception.GenreNotFoundException;
+import lt.techin.group.project.exception.MediaNotFoundException;
+import lt.techin.group.project.model.Genre;
+import lt.techin.group.project.model.Media;
+import lt.techin.group.project.repository.GenreRepository;
+import lt.techin.group.project.repository.MediaRepository;
+import lt.techin.group.project.rest.dto.MediaDto;
+import org.apache.coyote.BadRequestException;
+import org.springframework.stereotype.Service;
+
+import java.time.Year;
+import java.util.List;
+
+import static lt.techin.group.project.service.GenreService.GENRE_NOT_FOUND_WITH_ID;
+
+@Data
+@AllArgsConstructor
+@Service
```

Comment:
It's not critical, but here better to remove @Data annotation because it overloading class with generating getters, setters, toString, hashCode in background and holding it in spring app context.

Reply:
Solved, used Getter and Setter instead